packages/typespec-vscode/test/extension/common/commonSteps.ts

```diff
@@ -0,0 +1,355 @@
+      const contrastResult = page.getByText(new RegExp(text)).first()
+      return (await contrastResult.count()) > 0
+    },
+    errorMessage,
+    sleep
+  )
+}
+
+/**
+ * Results comparison
+ * @param res List of expected files
+ * @param dir The directory to be compared needs to be converted into an absolute path using path.resolve
+ */
+async function contrastResult(page: Page, res: string[], dir: string) {
+  let resLength = 0
+  const __dirname = path.dirname(fileURLToPath(import.meta.url));
+  if (fs.existsSync(dir)) {
+    resLength = fs.readdirSync(dir).length
+    fs.rmSync(path.resolve(__dirname, "../../images-linux"), { recursive: true, force: true })
+    fs.rmSync(dir, { recursive: true, force: true })
+  }
+  if (resLength !== res.length) {
+    await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/error.png") })
+    throw new Error("Failed to matches all files")
+  }
+}
+
+/**
+ * All cases need to execute the steps. Click the top input box and enter the command
+ * @param page vscode object
+ * @param {folderName, command}
+ * folderName: The text in the top input box is usually the current open root directory,
+ * command: After the top input box pops up, the command to be executed
+ */
+async function startWithCommandPalette(
+  page: Page,
+  { folderName, command }: { folderName: string; command: string }
+) {
+  await sleep(2)
+  await page.locator("li").filter({ hasText: folderName }).first().click()
+  await sleep(2)
+  const __dirname = path.dirname(fileURLToPath(import.meta.url));
+  await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/open_top_panel.png") })
+  await page
+    .getByRole("textbox", { name: "Search files by name (append" })
+    .first()
+    .fill(`>TypeSpec: ${command}`)
+  let listForCreate: Locator
+  await retry(
+    page,
+    5,
+    async () => {
+      listForCreate = page
+        .locator("a")
+        .filter({ hasText: `TypeSpec: ${command}` })
+        .first()
+      return (await listForCreate.count()) > 0
+    },
+    "Failed to find the specified option"
+  )
+  await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/input_command.png") })
+  await listForCreate!.click()
+}
+
+/**
+ * Start the Project with Right click on the file
+ * @param page vscode object
+ * @param command create, emit or import
+ * @param type specify whether the click is on file, folder or empty folder
+ * command: specify which command to execute to the project
+ */
+async function startWithRightClick(page: Page, command: string, type?: string) {
+  if (
+    command == "Emit from TypeSpec" ||
+    command == "Preview API Documentation"
+  ) {
+    const target = page.getByRole("treeitem", { name: "main.tsp" }).locator("a")
+    await target.click({ button: "right" })
+    const __dirname = path.dirname(fileURLToPath(import.meta.url));
+    await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/click_main.png") })
```

Comment:
would be good to have this images-linux path saved as a const on top or in a constant.ts file

Reply:
maybe make a 
```
function screenshot(page: PlaywrightPage, os: "linux", name: string);
```

Reply:
Updated to `function screenshot(page: Page, os: "linux", name: string)`.